rocky/reports/views/base.py

```diff
@@ -161,23 +142,34 @@ def get_ooi_filter_forms(self, ooi_types: Iterable[type[OOI]]) -> dict[str, Form
             )
         }
 
+    def ooi_selection_is_valid(self) -> bool:
+        is_valid = True
+        if not self.selected_oois:
+            is_valid = False
+            messages.error(self.request, _("Select at least one OOI to proceed."))
+        return is_valid
```

Comment:
This is a getter method with a side effect? The callee of the function should decide wether or not to throw the error based on the returned boolean

Reply:
Yes, the thing I want to accomplish here are 2 things. 
1. I want to keep 1 message central to use for all other views
2. I want to check the validity just to redirect at certain places. In some cases I just show the message, at other times I use. If it is not valid redirect to another url with the messaging. 

I cannot really find a good way to have both functionality in one, because just throwing the message will return a None. I cannot check for None values since that anything can be None even when the message is not thrown.

Reply:
or another possibility is to create a dictionary with error messaging to keep it central and use it where needed.

Reply:
Should I rename this function so it doesn't look like a plain getter function?

Reply:
@ammar92 

What do you think of this?

```
from enum import StrEnum

class OOISelectionError(StrEnum):
    none_selection = _("Select at least one OOI to proceed.")
```

We can add as many error messages so we an pull it where needed?

Reply:
You can use a constant that holds the message if it's shared between several functions or modules. A dictionary or enums of possible strings would make it worse

Reply:
fixed: https://github.com/minvws/nl-kat-coordination/blob/ea12f0edbbe3de17bc97ad95eb3dcc1416b39436/rocky/reports/views/aggregate_report.py#L83

Reply:
Such constants are usually put outside of the class and somewhere at the top in the module to make them easier to find. This way will improve code readability, because configuration values are separated from implementation logic.